packages/component-base/src/polylit-mixin.js

```diff
@@ -203,8 +224,13 @@ const PolylitMixinImplementation = (superclass) => {
         this.$ = {};
       }
 
-      this.renderRoot.querySelectorAll('[id]').forEach((node) => {
-        this.$[node.id] = node;
+      // Register all elements with an id in the $ map, including those
+      // that might have been already teleported to a different element
+      // (e.g. opened overlays are teleported to the body element).
+      [...Object.values(this.$), this.renderRoot].forEach((node) => {
+        node.querySelectorAll('[id]').forEach((node) => {
+          this.$[node.id] = node;
+        });
       });
```

Comment:
Is this change used by something? All tests in the changed packages pass without it

Reply:
I added it to make sure that any potential child elements with IDs in the light DOM of teleported components are also registered, but it looks like no such cases exist in our codebase, so this change appears to be unnecessary after all. Let me revert it.

Reply:
I turns out it was needed. Login tests fail without it: https://github.com/vaadin/web-components/actions/runs/12271322766/job/34238003066?pr=8290

Reply:
Done: https://github.com/vaadin/web-components/pull/8321